src/services/navigateTo.ts

```diff
@@ -1,35 +1,42 @@
 module ts.NavigateTo {
-    type RawNavigateToItem = { name: string; fileName: string; matchKind: MatchKind; declaration: Declaration };
-
-    enum MatchKind {
-        none = 0,
-        exact = 1,
-        substring = 2,
-        prefix = 3
-    }
-
-    export function getNavigateToItems(program: Program, cancellationToken: CancellationTokenObject, searchValue: string, maxResultCount: number): NavigateToItem[]{
-        // Split search value in terms array
-        var terms = searchValue.split(" ");
-
-        // default NavigateTo approach: if search term contains only lower-case chars - use case-insensitive search, otherwise switch to case-sensitive version
-        var searchTerms = map(terms, t => ({ caseSensitive: hasAnyUpperCaseCharacter(t), term: t }));
+    type RawNavigateToItem = { name: string; fileName: string; matchKind: PatternMatchKind; isCaseSensitive: boolean; declaration: Declaration };
 
+    export function getNavigateToItems(program: Program, cancellationToken: CancellationTokenObject, searchValue: string, maxResultCount: number): NavigateToItem[] {
+        var patternMatcher = createPatternMatcher(searchValue);
         var rawItems: RawNavigateToItem[] = [];
 
         // Search the declarations in all files and output matched NavigateToItem into array of NavigateToItem[] 
         forEach(program.getSourceFiles(), sourceFile => {
             cancellationToken.throwIfCancellationRequested();
 
-            var fileName = sourceFile.fileName;
             var declarations = sourceFile.getNamedDeclarations();
             for (var i = 0, n = declarations.length; i < n; i++) {
                 var declaration = declarations[i];
-                // TODO(jfreeman): Skip this declaration if it has a computed name
-                var name = (<Identifier>declaration.name).text;
-                var matchKind = getMatchKind(searchTerms, name);
-                if (matchKind !== MatchKind.none) {
-                    rawItems.push({ name, fileName, matchKind, declaration });
+                var name = getDeclarationName(declaration);

```

Comment:
This will crash on a computed property that is not a well known symbol. Either check for that first, or remove the assert inside.